subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -513,47 +520,76 @@ int fota_download_start_with_image_type(const char *host, const char *file,
 	return 0;
 }
 
-int fota_download_init(fota_download_callback_t client_callback)
+static int fota_download_object_init(void)
+{
+	int err;
+
+#ifdef CONFIG_FOTA_DOWNLOAD_NATIVE_TLS
```

Comment:
Can you use `if (IS_ENABLED(...` here instead? that is always preferable in situations like these - assuming that you don't et compiler issues.

Reply:
OK This is originally this but I can now change that format.